dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java

```diff
@@ -5336,4 +5341,757 @@ public void JwtBuilderAPIBasicTests_multiple_apis_and_consumeToken() throws Exce
+        // give it a flag that says jwt string, and then pass the 3 part jwt token string
+        JSONObject testSettings = new JSONObject();
+        testSettings.put(JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM, JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM_JWT_STRING);
+        testSettings.put(JWTBuilderConstants.JWT_TOKEN, jwtToken);
+
+        // Now, add any override values - for this test, there are none
+        // Invoke the builder app to create a token (using (a second) builder, load all claims from the token into the second builder
+        String builderId = "jwt1";
+        Page response = actions.invokeJwtBuilder_setApis(_testName, builderServer, builderId, testSettings);
+
+        validationUtils.validateResult(response, expectations);
+
+    }
+
+    // TODO Failing since claimsFrom doesn't know how to parse JWE Token
+    //@Test
+    public void JwtBuilderAPIBasicTests_claimsFrom_jweStringRS256_to_jweTokenRS384_allClaims() throws Exception {
+
+        // build a token using the alternate builder config (we'll get claims from it)
+        String baseBuilderId = "encryptJwtRS256";
+        String jwtToken = getBaseToken(baseBuilderId);
+        JSONObject baseSettings = BuilderHelpers.setClaimsFromToken(jwtToken, JwtKeyTools.getComplexPrivateKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256));
+        // second token will be encrypted - we don't have a way to pass the key to the validation methods, so, we'll skip the checks that required the actual payload content
+        baseSettings.remove(PayloadConstants.EXPIRATION_TIME);
+        baseSettings.remove(PayloadConstants.ISSUED_AT);
+        // since we getting all claims in the case, we can base our expectations upon the content of this first token)
+        Expectations expectations = BuilderHelpers.createGoodBuilderExpectations(JWTBuilderConstants.JWT_BUILDER_SETAPIS_ENDPOINT, baseSettings, builderServer);
+
+        // build settings that will tell the test app how to run/what to pass to the "claimFrom" api
+        // give it a flag that says jwt string, and then pass the 3 part jwt token string
+        JSONObject testSettings = new JSONObject();
+        testSettings.put(JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM, JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM_JWT_STRING);
+        testSettings.put(JWTBuilderConstants.JWT_TOKEN, jwtToken);
+
+        // Now, add any override values - for this test, there are none
+        // Invoke the builder app to create a token (using (a second) builder, load all claims from the token into the second builder
+        String builderId = "encryptJwtRS384";
+        Page response = actions.invokeJwtBuilder_setApis(_testName, builderServer, builderId, testSettings);
+
+        validationUtils.validateResult(response, expectations);
+
+    }
+
+    @Test
+    public void JwtBuilderAPIBasicTests_claimsFrom_jweString_encodedPayload_allClaims() throws Exception {
```

Comment:
Is this not doing the same thing as `JwtBuilderAPIBasicTests_claimsFrom_jwtString_to_jweToken_allClaims`?

Reply:
no - this is taking a JWE and pulling the payload out (in the test code) and passing just the payload (it's really the same as using a JWS payload test that we have, just making sure that the payload was formed properly when it comes from a JWE.